clio-server/src/main/scala/org/broadinstitute/clio/server/ClioServer.scala

```diff
@@ -67,23 +62,36 @@ object ClioServer
     new ClioApp(
       serverStatusDAO,
       auditDAO,
-      httpServerDAO,
       persistenceDAO,
       searchDAO
     )
 
-  private val serverService = ServerService(app)
   private val persistenceService = PersistenceService(app)
   private val searchService = SearchService(app)
   override val auditService = AuditService(app)
-  override val statusService = StatusService(app)
-
-  override val wgsUbamService =
-    new WgsUbamService(persistenceService, searchService)
-  override val gvcfService =
-    new GvcfService(persistenceService, searchService)
-  override val wgsCramService =
-    new WgsCramService(persistenceService, searchService)
+
+  val wgsUbamWebService =
+    new WgsUbamWebService(
+      new WgsUbamService(persistenceService, searchService)
+    )
+
+  val gvcfWebService =
+    new GvcfWebService(
+      new GvcfService(persistenceService, searchService)
+    )
+
+  val wgsCramWebService =
+    new WgsCramWebService(
+      new WgsCramService(persistenceService, searchService)
+    )
+
+  private val apiRoutes: Route =
+    concat(wgsUbamWebService.routes, gvcfWebService.routes, wgsCramWebService.routes)
+
+  private val httpServerDAO = AkkaHttpServerDAO(wrapperDirectives, infoRoutes, apiRoutes)
+
+  private val serverService = ServerService(app, httpServerDAO)
+  override val statusService = StatusService(app, httpServerDAO)
```

Comment:
Unrelated, but if we're waiting to merge this... could you make whatever this is `override`-ing a class instead of a trait, then add it in through composition? Similar to what you've done with the web services. It makes setting up an isolated instance for testing much easier.

Reply:
Not sure what you mean by this. ClioServer extends a JsonWebService with several traits. If the StatusWebService trait that this val comes from is an abstract Class.. then in order to preserve everything else, it would have to extend a JsonWebService. Is that what you mean? I also don't entirely understand why it's an override in the first place, since it's just an abstract def in the trait

Reply:
I mean, make `ClioServer` no longer extend any of these traits, and instead have it include one member of each. So it'd have a `StatusWebService` member, similar to how it now has a `wgsUbamWebService` member. Then we don't have to go through all the headaches of making sure the names of fields within the traits don't clash, and it's easier to instantiate a stand-alone version of each web-service for testing.

`JsonWebService` can still be the parent trait of all the individual web-services, but it shouldn't be mixed into `ClioServer` any more.

(This could be another chunky refactor on its own; just a suggestion for something to tear into if you're waiting on the other refactor to merge)

Reply:
Were you thinking all of the Directives that are mixed in to ClioServer as well? I've done it for StatusWebService... but it involved a little funkiness with lazification and forward referencing